### PR TITLE
app-project: Refactor Announcement banners styling and remove artsy/fresnel

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
app-project

## Linked Issue and/or Talk Post
Toward https://github.com/zooniverse/front-end-monorepo/issues/2029 and a request from Sean to improve the styling of the banners by allowing contents to fill the viewport width.

## Describe your changes
- Refactor GenericAnnouncement to have one component and one style. There's no need for separate components or styling rules to display on desktop widths or mobile widths.
- Remove `@artsy/fresnel` package. This was a third-party intended to help with SSR styling, but banners was the only place it was used. I also have a strong opinion that the Media component in app-project was easily confused with Media we use frequently from lib-react-components.
- Remove `.defaultProps` from GenericAnnouncement.
- Refactor ProjectAnnouncement's unit tests to remove enzyme.

## How to Review

- All of app-project's Announcements have their own component in Storybook. 
- i-fancy-cats has a banner, and here's an example of a live FEM project with a couple of banners: https://local.zooniverse.org:3000/projects/mariaedgeworthletters/maria-edgeworth-letters?env=production
- Once merged, I'll get a double-check on design review from Sean via frontend.preview.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/app-project/pages/_app.js [hunk 1]
```diff
@@ -104,19 +103,17 @@ function MyApp({ Component, pageProps }) {
         <GlobalStyle />
         <Provider store={store}>
           <ThemeModeContext.Provider value={themeContext}>
-            <MediaContextProvider disableDynamicMediaQueries>
```
Comment (goplayoutside3) [inline packages/app-project/pages/_app.js:107]: This was only used for GenericAnnouncement, so it's safe to remove from `_app.js` and `_document.js`

File: packages/app-project/src/components/Announcements/components/GenericAnnouncement/markdownzComponents.js [hunk 2]
```diff
@@ -3,11 +3,11 @@ import styled from 'styled-components'
 
 const StyledParagraph = styled(Paragraph)`
   font-weight: bold;
-  max-width: 60em;
+  line-height: 1.25;
```
Comment (goplayoutside3) [inline packages/app-project/src/components/Announcements/components/GenericAnnouncement/markdownzComponents.js:6]: Styling adjustment requested by Sean.

File: packages/app-project/src/screens/ProjectHomePage/components/Hero/Hero.stories.js [hunk 3]
```diff
@@ -91,16 +90,14 @@ const snapshot = {
   project: PROJECT,
   user: USER
 }
-const store = Store.create(snapshot)  
+const store = Store.create(snapshot)
 
 export function Default({ isWide }) {
   return (
     <RouterMock>
-      <MediaContextProvider>
```
Comment (goplayoutside3) [inline packages/app-project/src/screens/ProjectHomePage/components/Hero/Hero.stories.js:99]: <Media> hasn't been used in Hero for at least a year, so this was just lingering in its Storybook file.

**Review by shaunanoordin — APPROVED** [review]
## PR Overview

Package: app-project

This PR updates the code for the Announcement Banners on project pages. There are minor visual changes, but **no** behaviour changes intended.

_Banners on a project, 1000px wide_

image

_Banners on a project, 400px wide_

image

Visual changes:
- The distinction between "wide" view and "narrow" view has been removed.

No behaviour changes:
- Announcement banners (`<GenericAnnouncement>`s) appear when:
  - Project is finished, and includes a link to Results if any (green background, tested)
  - Project has a custom announcement, and can be dismissed (yellow background, tested)
  - User is invited to sign in/register (not tested - not sure what the trigger is)

Notable code changes:
- artsy/fresnel removed, meaning the `<Media>` shared component no longer needs to exist.
- WideGenericAnnouncement and NarrowGenericAnnouncement now completely folded into GenericAnnouncement.

### Testing

Tested with macOS + Chrome 128, on app-project, localhost

- Viewed a project with a Finished banner and a Project Announcement banner: https://local.zooniverse.org:3000/projects/mariaedgeworthletters/maria-edgeworth-letters?env=production
- Confirmed content is visible on both on wide screen and narrow screens.
- Confirmed that "See Results" link works.
- Confirmed that 'X' button closes Project Announcement banner.

### Status

Code changes look legit, and functionality checks out. LGTM 👍

**Comment by eatyourgreens** [comment]
This PR regressed #2471.

**Comment by goplayoutside3** [comment]
@eatyourgreens please provide a link to production of a Zooniverse project where this is occuring, or a PR using Storybook to fix.

**Comment by eatyourgreens** [comment]
`am.zooni` posted a few links to broken banners on Talk: https://www.zooniverse.org/talk/17/3440569?comment=5651819&page=1

**Comment by goplayoutside3** [comment]
From here post: https://www.zooniverse.org/projects/md68135/notes-from-nature-labs and https://www.zooniverse.org/projects/bmtcollections/documentation-detectives-transcribing-accession-registers

**Pull request merged**